### ISSUE TITLE
Train on validation set?

### ISSUE THREAD

**Issue by bkj** [gyhui14/spottune#4]
I notice here on this line:

https://github.com/gyhui14/spottune/blob/master/imdbfolder.py#L97

it looks like you're training on both the training and validation set, and validating on the validation set.  Is this intentional?  It makes sense if you're trying to get maximum accuracy on the test set, but of course means that the validation numbers you get during training are basically meaningless. 

I tracked down the bug because I noticed that my validation accuracy was going to 0.99 for `aircraft`, which was way off the test accuracies reported in the paper.

Thoughts?

Thanks!

__Edit:__ If I leave the code as-is and train using the validation data, I get an accuracy of ~ 0.66 on `aircraft`.  If I train on the train data only, I get an accuracy of ~0.50.  So that suggests that the results on the paper _did_ train on the validation set -- which is fine, so long as everyone else that's submitting is doing the same.  Do we know if they are?

**Comment by gyhui14** [comment]
Hi bkj, thanks for your interest. Yeah, that's intentional. We tested the previous state-of-the-art approach (serial residual adapter) and if it was only trained on the train set (using the code of the author), the test accuracy on aircraft is around 0.327 which is much lower the result in the paper (0.618). And the training from scratch baseline on aircraft achieves 0.571. The parallel residual adapter with SVD achieves 66.04. It suggests that previous approaches (at least residual adapters) are trained on both the train set and val set.  Thanks.

**Comment by YuWang24** [comment]
> 
> 
> I notice here on this line:
> 
> https://github.com/gyhui14/spottune/blob/master/imdbfolder.py#L97
> 
> it looks like you're training on both the training and validation set, and validating on the validation set. Is this intentional? It makes sense if you're trying to get maximum accuracy on the test set, but of course means that the validation numbers you get during training are basically meaningless.
> 
> I tracked down the bug because I noticed that my validation accuracy was going to 0.99 for `aircraft`, which was way off the test accuracies reported in the paper.
> 
> Thoughts?
> 
> Thanks!
> 
> **Edit:** If I leave the code as-is and train using the validation data, I get an accuracy of ~ 0.66 on `aircraft`. If I train on the train data only, I get an accuracy of ~0.50. So that suggests that the results on the paper _did_ train on the validation set -- which is fine, so long as everyone else that's submitting is doing the same. Do we know if they are?

Hi bkj and Dr Guo

I am very confused that the reported accuracy for aircraft is 63.91% in the paper. However, when I run the code and train on the aircraft, I got an accuracy of ~0.99 as bkj stated in his original post. What could be the cause of this situation? How does bkj fix it? 
I am running it using Python 3.6, this should be the only difference, other environments are all the same. Any suggestions will be appreciated. Thanks.

**Comment by gyhui14** [comment]
Hi YuWang24, to get the test accuracy, you need to submit the results.json to the visual decathlon challenge website.

**Comment by gyhui14** [comment]
> > I notice here on this line:
> > https://github.com/gyhui14/spottune/blob/master/imdbfolder.py#L97
> > it looks like you're training on both the training and validation set, and validating on the validation set. Is this intentional? It makes sense if you're trying to get maximum accuracy on the test set, but of course means that the validation numbers you get during training are basically meaningless.
> > I tracked down the bug because I noticed that my validation accuracy was going to 0.99 for `aircraft`, which was way off the test accuracies reported in the paper.
> > Thoughts?
> > Thanks!
> > **Edit:** If I leave the code as-is and train using the validation data, I get an accuracy of ~ 0.66 on `aircraft`. If I train on the train data only, I get an accuracy of ~0.50. So that suggests that the results on the paper _did_ train on the validation set -- which is fine, so long as everyone else that's submitting is doing the same. Do we know if they are?
> 
> Hi bkj and Dr Guo
> 
> I am very confused that the reported accuracy for aircraft is 63.91% in the paper. However, when I run the code and train on the aircraft, I got an accuracy of ~0.99 as bkj stated in his original post. What could be the cause of this situation? How does bkj fix it?
> I am running it using Python 3.6, this should be the only difference, other environments are all the same. Any suggestions will be appreciated. Thanks.

Hi YuWang24, to get the test accuracy, you need to submit the results.json to the visual decathlon challenge website.

**Comment by YuWang24** [comment]
> 
> 
> > > I notice here on this line:
> > > https://github.com/gyhui14/spottune/blob/master/imdbfolder.py#L97
> > > it looks like you're training on both the training and validation set, and validating on the validation set. Is this intentional? It makes sense if you're trying to get maximum accuracy on the test set, but of course means that the validation numbers you get during training are basically meaningless.
> > > I tracked down the bug because I noticed that my validation accuracy was going to 0.99 for `aircraft`, which was way off the test accuracies reported in the paper.
> > > Thoughts?
> > > Thanks!
> > > **Edit:** If I leave the code as-is and train using the validation data, I get an accuracy of ~ 0.66 on `aircraft`. If I train on the train data only, I get an accuracy of ~0.50. So that suggests that the results on the paper _did_ train on the validation set -- which is fine, so long as everyone else that's submitting is doing the same. Do we know if they are?
> > 
> > 
> > Hi bkj and Dr Guo
> > I am very confused that the reported accuracy for aircraft is 63.91% in the paper. However, when I run the code and train on the aircraft, I got an accuracy of ~0.99 as bkj stated in his original post. What could be the cause of this situation? How does bkj fix it?
> > I am running it using Python 3.6, this should be the only difference, other environments are all the same. Any suggestions will be appreciated. Thanks.
> 
> Hi YuWang24, to get the test accuracy, you need to submit the results.json to the visual decathlon challenge website.

Hi Dr Guo
Thanks for your quick reply. So, does that mean the Table 3 in the paper are all test accuracies? But, why the validation accuracy is so high around 0.99? Could you please offer any rough explanations? Thanks.

**Comment by gyhui14** [comment]
> > > > I notice here on this line:
> > > > https://github.com/gyhui14/spottune/blob/master/imdbfolder.py#L97
> > > > it looks like you're training on both the training and validation set, and validating on the validation set. Is this intentional? It makes sense if you're trying to get maximum accuracy on the test set, but of course means that the validation numbers you get during training are basically meaningless.
> > > > I tracked down the bug because I noticed that my validation accuracy was going to 0.99 for `aircraft`, which was way off the test accuracies reported in the paper.
> > > > Thoughts?
> > > > Thanks!
> > > > **Edit:** If I leave the code as-is and train using the validation data, I get an accuracy of ~ 0.66 on `aircraft`. If I train on the train data only, I get an accuracy of ~0.50. So that suggests that the results on the paper _did_ train on the validation set -- which is fine, so long as everyone else that's submitting is doing the same. Do we know if they are?
> > > 
> > > 
> > > Hi bkj and Dr Guo
> > > I am very confused that the reported accuracy for aircraft is 63.91% in the paper. However, when I run the code and train on the aircraft, I got an accuracy of ~0.99 as bkj stated in his original post. What could be the cause of this situation? How does bkj fix it?
> > > I am running it using Python 3.6, this should be the only difference, other environments are all the same. Any suggestions will be appreciated. Thanks.
> > 
> > 
> > Hi YuWang24, to get the test accuracy, you need to submit the results.json to the visual decathlon challenge website.
> 
> Hi Dr Guo
> Thanks for your quick reply. So, does that mean the Table 3 in the paper are all test accuracies? But, why the validation accuracy is so high around 0.99? Could you please offer any rough explanations? Thanks.

Hi YuWang24, that's because of the validation data is combined with the training data.

**Comment by YuWang24** [comment]
Oh, okay I get it. By the way, is there any ways to get the test accuracy without submitting the result to visual decathlon challenge website?

**Comment by gyhui14** [comment]
> Oh, okay I get it. By the way, is there any ways to get the test accuracy without submitting the result to visual decathlon challenge website?

I think that's the only way to get the test results.
